services/src/main/java/org/keycloak/protocol/oidc/grants/ciba/clientpolicy/executor/SecureCibaAuthenticationRequestSigningAlgorithmExecutor.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.protocol.oidc.grants.ciba.clientpolicy.executor;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.jboss.logging.Logger;
+
+import org.keycloak.OAuthErrorException;
+import org.keycloak.crypto.Algorithm;
+import org.keycloak.models.CibaConfig;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.representations.idm.ClientPolicyExecutorConfigurationRepresentation;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.services.clientpolicy.ClientPolicyContext;
+import org.keycloak.services.clientpolicy.ClientPolicyException;
+import org.keycloak.services.clientpolicy.context.AdminClientRegisterContext;
+import org.keycloak.services.clientpolicy.context.AdminClientUpdateContext;
+import org.keycloak.services.clientpolicy.context.DynamicClientRegisterContext;
+import org.keycloak.services.clientpolicy.context.DynamicClientUpdateContext;
+import org.keycloak.services.clientpolicy.executor.ClientPolicyExecutorProvider;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * @author <a href="mailto:takashi.norimatsu.ws@hitachi.com">Takashi Norimatsu</a>
+ */
+public class SecureCibaAuthenticationRequestSigningAlgorithmExecutor implements ClientPolicyExecutorProvider<SecureCibaAuthenticationRequestSigningAlgorithmExecutor.Configuration> {
+
+    private static final Logger logger = Logger.getLogger(SecureCibaAuthenticationRequestSigningAlgorithmExecutor.class);
+
+    private final KeycloakSession session;
+    private Configuration configuration;
+
+    private static final String sigTarget = CibaConfig.CIBA_BACKCHANNEL_AUTH_REQUEST_SIGNING_ALG;
+
+    private static final String DEFAULT_ALGORITHM_VALUE = Algorithm.PS256;
+
+    static final Set<String> ALLOWED_ALGORITHMS = new LinkedHashSet<>(Arrays.asList(
```

Comment:
I wonder if we can introduce the shared constant somewhere as this list of FAPI secured algorithms is repeated on few places. How about introduce the `FapiConstant` class and add the ALLOWED_ALGORITHMS constant there? This will allow that it can be consumed from all places (SecureCibaAuthenticationRequestSigningAlgorithmExecutor, SecureSigningAlgorithmExecutor, SecureSigningAlgorithmForSignedJwtExecutor and maybe more in the future if needed...).

That way, if there are more FAPI supported algorithms in the future, we can change it in single place in the constant. WDYT?

Reply:
@mposolda Yes, I will add `FapiConstant` as you've proposed and put FAPI allowing signature algorithm into this single place.